src/App.tsx

```diff
@@ -1,61 +1,113 @@
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import { getUserById } from './services/user';
+import { Todo } from './types/Todo';
+import todosFromServer from './api/todos';
+import { ChangeEvent, useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const initialTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+const maxId = todosFromServer.reduce((max, todo) => {
+  return todo.id > max ? todo.id : max;
+}, todosFromServer[0].id);
+
+export const App: React.FC = () => {
+  const [id, setId] = useState(maxId);
+  const [todos, setTodos] = useState<Todo[]>(initialTodos);
+  const onSubmit = (newTodo: Todo) => {
+    setTodos(currentTodos => [...currentTodos, newTodo]);
+  };
+
+  const users = [...usersFromServer];
```

Comment:
Unnecessarily creating a variable that changes every re-render; why not simply use `usersFromServer` or name it `users` when importing?